### PR TITLE
Domains: Fix "Go Back" Header for Add new transfer

### PR DESCRIPTION
**Pull request by aidvu**
Go Back on add new transfer screen would always send you
back to domain search. It should return you to the add domain
for transfer screen if you were on precheck.

Also fix Map It screen button width as it was shown as "A.."

Test:
- Go to Add Domain
- Use a domain I own
- Try adding your domain
- On pre-check screen click back
- Make sure you're back on domain input for transfer
- Click back again
- Make sure you're on add domain screen

Fixes: https://github.com/Automattic/wp-calypso/issues/19912

### CODE REVIEW

File: client/components/formatted-header/index.jsx [hunk 1]
```diff
@@ -29,7 +29,7 @@ function FormattedHeader( { headerText, subHeaderText } ) {
 }
 
 FormattedHeader.propTypes = {
-	headerText: PropTypes.string,
+	headerText: PropTypes.oneOfType( [ PropTypes.string, PropTypes.array ] ),
```
Comment (klimeryk) [inline client/components/formatted-header/index.jsx]: [`prop-types` have a `node` type](https://github.com/facebook/prop-types), too - which might be a better fit here:
```
  // Anything that can be rendered: numbers, strings, elements or an array
  // (or fragment) containing these types.
```

Comment (aidvu) [reply client/components/formatted-header/index.jsx]: Smart. :)

**Comment by aidvu** [comment]
@klimeryk not really sure for mapping. It uses a direct link, so fairly hard. :(

**Comment by klimeryk** [comment]
> @klimeryk not really sure for mapping. It uses a direct link, so fairly hard. :(

I believe in you 😜

File: client/components/formatted-header/index.jsx (hunk 1 above)
Comment (aidvu) [reply client/components/formatted-header/index.jsx]: Actually, after looking at it, string/array makes more sense. Narrows it down, and it's actual text, just that `translate` returns an array.

**Comment by aidvu** [comment]
@klimeryk it's a global thing on how we do navigation. If we use direct links, would need to pass params for it, and makes little sense, but sure, can do that later. 😄

**Pull request merged**